File: 0090-subsets-ii/0090-subsets-ii.cpp
class Solution {
public:
    void check(vector<int>&nums , vector<vector<int>>&v, int idx , vector<int>temp){
        v.push_back(temp) ; 

        for(int i = idx ; i<nums.size(); i++){
            if(i>idx && nums[i] == nums[i-1] ) continue ;
            temp.push_back(nums[i]) ; 
            check(nums , v , i+1 , temp) ; 
            temp.pop_back();
        }
    }
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        sort(nums.begin() , nums.end()) ; 
        vector<vector<int>>v ; 
        vector<int>temp  ;
        check(nums , v , 0 , temp ) ; 
        return v ;
    }
};

File: 0090-subsets-ii/0090-subsets-ii.java
class Solution {
    void check(int[] nums , int idx , List<Integer>temp , List<List<Integer>>v){
        v.add(new ArrayList<>(temp)); 

        for(int i = idx ; i<nums.length ; i++){
            if(i>idx && nums[i] == nums[i-1]) continue ;
            temp.add(nums[i]); 
            check(nums , i + 1 , temp , v) ; 
            temp.remove(temp.size()-1) ; 
        }
    }
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums) ; 
        List<List<Integer>>ans = new ArrayList<>() ; 
        List<Integer>temp = new ArrayList<>() ; 

        check(nums , 0 , temp , ans) ; 

        return ans ;
    }
}